client/core/account_test.go

```diff
@@ -88,6 +90,93 @@ func setupRigAccountProof(host string, rig *testRig) {
 	rig.db.accountProof = accountProof
 }
 
+func TestAccountDisable(t *testing.T) {
+	activeTrades := map[order.OrderID]*trackedTrade{
+		order.OrderID{}: &trackedTrade{metaData: &db.OrderMetaData{Status: order.OrderStatusBooked}},
+	}
+
+	tests := []struct {
+		name, host                          string
+		recryptErr, acctErr, disableAcctErr error
+		wantErr, wantErrCode, loseConns     bool
+		activeTrades                        map[order.OrderID]*trackedTrade
+		errCode                             int
+	}{{
+		name: "ok",
+		host: tDexHost,
+	}, {
+		name:       "password error",
+		host:       tDexHost,
+		recryptErr: tErr,
+		wantErr:    true,
+		errCode:    passwordErr,
+	}, {
+		name:        "host error",
+		host:        ":bad:",
+		wantErr:     true,
+		wantErrCode: true,
+		errCode:     addressParseErr,
+	}, {
+		name:        "dex not in conns",
+		host:        tDexHost,
+		loseConns:   true,
+		wantErr:     true,
+		wantErrCode: true,
+		errCode:     unknownDEXErr,
+	}, {
+		name:         "has active orders",
+		host:         tDexHost,
+		activeTrades: activeTrades,
+		wantErr:      true,
+	}, {
+		name:        "retrieve account error",
+		host:        tDexHost,
+		acctErr:     errors.New(""),
+		wantErr:     true,
+		wantErrCode: true,
+		errCode:     accountRetrieveErr,
+	}, {
+		name:           "disable account error",
+		host:           tDexHost,
+		disableAcctErr: errors.New(""),
+		wantErr:        true,
+		wantErrCode:    true,
+		errCode:        accountDisableErr,
+	}}
+
+	for _, test := range tests {
+		rig := newTestRig()
+		tCore := rig.core
```

Comment:
It's good practice to call `rig.shutdown` at some point. Ideally, that would be at the end of the subtest, but at the end of `TestAccountDisable` is better than nothing.